### PR TITLE
Sovereign header sig verifier

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- Added possibility to verify extra signatures from header in header sig verifier 
  
## Proposed changes
- An extra header sig verifier is added
- Each extra verifier is registered in a `holder` of vierfiers. Basically you can add any number of extra custom verifiers.
- Fixed a bug in case there is no outgoing operation

## Testing procedure
image

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by sasurobert — DISMISSED** [review]

File: consensus/spos/bls/sovereignSubRoundEndOutGoingTxData.go [hunk 1]
```diff
@@ -43,8 +43,18 @@ func (sr *sovereignSubRoundEndOutGoingTxData) VerifyAggregatedSignatures(bitmap
 }
 
 // AggregateAndSetSignatures aggregates and sets signatures for outgoing tx data
-func (sr *sovereignSubRoundEndOutGoingTxData) AggregateAndSetSignatures(bitmap []byte, epoch uint32) ([]byte, error) {
-	sig, err := sr.signingHandler.AggregateSigs(bitmap, epoch)
+func (sr *sovereignSubRoundEndOutGoingTxData) AggregateAndSetSignatures(bitmap []byte, header data.HeaderHandler) ([]byte, error) {
```
Comment (raduchis) [inline consensus/spos/bls/sovereignSubRoundEndOutGoingTxData.go:46]: is it necessary to have all of these files in the spos/bls? they do not seem to implement SubRoundHandler

File: consensus/spos/bls/sovereignSubRoundSignatureOutGoingTxData.go [hunk 2]
```diff
@@ -50,12 +50,18 @@ func (sr *sovereignSubRoundSignatureOutGoingTxData) CreateSignatureShare(
...
 }
 
 // StoreSignatureShare stores the provided sig share for outgoing tx data from the consensus message
 func (sr *sovereignSubRoundSignatureOutGoingTxData) StoreSignatureShare(index uint16, cnsMsg *consensus.Message) error {
-	return sr.signingHandler.StoreSignatureShare(index, cnsMsg.SignatureShareOutGoingTxData)
+	if cnsMsg != nil && len(cnsMsg.SignatureShareOutGoingTxData) != 0 {
+		return sr.signingHandler.StoreSignatureShare(index, cnsMsg.SignatureShareOutGoingTxData)
+	}
+
+	return nil
```
Comment (raduchis) [inline consensus/spos/bls/sovereignSubRoundSignatureOutGoingTxData.go:64]: should no error be returned if the cnsMsg is nil?

File: testscommon/headerSigVerifier/extraHeaderSigVerifierHolderStub.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
+package headerSigVerifier
+
+import (
+	"github.com/multiversx/mx-chain-core-go/data"
+	crypto "github.com/multiversx/mx-chain-crypto-go"
+	"github.com/multiversx/mx-chain-go/process"
+)
+
+// ExtraHeaderSigVerifierHolderMock -
+type ExtraHeaderSigVerifierHolderMock struct {
```
Comment (raduchis) [inline testscommon/headerSigVerifier/extraHeaderSigVerifierHolderStub.go:10]: filename should be with Mock

File: consensus/spos/bls/sovereignSubRoundSignatureOutGoingTxData.go (hunk 2 above)
Comment (mariusmihaic) [reply consensus/spos/bls/sovereignSubRoundSignatureOutGoingTxData.go:64]: Right, returned error

File: consensus/spos/bls/sovereignSubRoundEndOutGoingTxData.go (hunk 1 above)
Comment (mariusmihaic) [reply consensus/spos/bls/sovereignSubRoundEndOutGoingTxData.go:46]: Yes, better to keep them all together.

Comment (raduchis) [reply consensus/spos/bls/sovereignSubRoundEndOutGoingTxData.go:46]: ok

**Pull request merged**